### PR TITLE
Update syntax for type annotations

### PR DESCRIPTION
**Pull request by pierwill**
## Status

Ready for review

## Description of Changes

Use PEP484 syntax in ./securedrop/rm.py.

Work toward #5282.

## Testing

- [ ] `make typelint` exits zero

### CODE REVIEW

File: securedrop/rm.py [hunk 1]
```diff
@@ -22,8 +22,7 @@
 import subprocess
 
 
-def shred(path, delete=True):
-    # type: (str, bool) -> None
+def shred(path: str, delete: bool=True) -> None:
```
Comment (kushaldas) [inline securedrop/rm.py:25]: This should be

```
def shred(path: str, delete: bool = True) -> None:
```

**Comment by pierwill** [comment]
Thanks! Force-pushed that change—

**Comment by kushaldas** [comment]
We will have to rebase after https://github.com/freedomofpress/securedrop/pull/5286 is merged.

**Comment by rmol** [comment]
This likely just needs one more rebase, because of the Tor browser update.

**Comment by kushaldas** [comment]
@pierwill let CI get done, I will re-review formally and merge. Thank you once again :)

**Pull request merged**